lib/github/ldap.rb

```diff
@@ -193,6 +196,26 @@ def search(options, &block)
       end
     end
 
+    # Internal: Search within a ldap forest
+    #
+    # Returns an Array of Net::LDAP::Entry.
+    def forest_search(options, &block)
+      instrument "forest_search.github_ldap" do |payload|
+        result =
+          if @forest.empty?
+            @connection.search(options, &block)
+          else
+            @forest.each_with_object([]) do |(rootdn, server), res|
+              if options[:base].end_with?(rootdn)
```

Comment:
Requires to use ```end_with?()``` and not ```eql?()``` which will result in unnecessary requests if you have a setup like ```DC=foo,DC=bar,DC=com``` & ```DC=bar,DC=com```. Issue is that calls to the library contain requests where the basedn points to the object of interest.
* Login for example will query the full path: ```options[:base]='CN=user,OU=org,DC=foo,DC=bar,DC=com'``` to get the user.
* Possible fix would be to form such queries (outside the library) to search for the distinguished name: ```options[:filter]='distinguishedName=CN=user,OU=org,DC=foo,DC=bar,DC=com'``` with a basedn ```options[:base]='DC=foo,DC=bar,DC=com'```